internal/internal_task_handlers.go

```diff
@@ -1586,16 +1589,27 @@ func (wth *workflowTaskHandlerImpl) completeWorkflow(
 		// Continue as new error.
 		metricsHandler.Counter(metrics.WorkflowContinueAsNewCounter).Inc(1)
 		closeCommand = createNewCommand(enumspb.COMMAND_TYPE_CONTINUE_AS_NEW_WORKFLOW_EXECUTION)
+
+		useCompat := true
```

Comment:
I see this logic repeated three times.. factor it out?

Reply:
To be clear the reason I didn't start it that way is that the defaults could, in principle, change on a per-command basis. But, if/when that ever happens then the shared thing can become un-shared so it's fine, changed this.